steward/src/utils.rs

```diff
@@ -32,3 +37,27 @@ pub async fn get_delegates_keys_by_orchestrator(
 
     Ok(keys)
 }
+
+fn check_scheme(input: &Url, original_string: &str) -> Result<(), Error> {
```

Comment:
Is this not something that would be captured in an error from `Provider::<Http>::try_from`?